http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -820,7 +822,22 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       appOpt,
       AppNotFoundByWorkspaceIdException(workspaceId, appName, ctx.traceId, "No active app found in DB")
     )
-    _ <- deleteAppV2Base(appResult.app, userInfo, workspaceId, deleteDisk)
+
+    workspaceApi <- wsmClientProvider.getWorkspaceApi(userInfo.accessToken.token)
+    attempt <- F.delay(workspaceApi.getWorkspace(workspaceId.value, IamRole.READER)).attempt
+
+    _ <- attempt match {
+      // if the workspace is found, delete the app normally
+      case Right(workspaceDesc) =>
+        deleteAppV2Base(appResult.app, appResult.cluster.cloudContext, userInfo, workspaceId, deleteDisk, workspaceDesc)
+      // if the workspace can't be found, delete the workspace records
+      case Left(_: NotFoundException) =>
```

Comment:
One thing to double check though is that the `NotFoundException` does not also corresponds with a Permission Denied. 

I know that internally within Leo we often return a RuntimeNotFound when a user does not have permission for instance